### ISSUE TITLE
Pop up virtual keyboard upon clicking on "Show Chat" on Deck

### ISSUE THREAD

**Issue by NyaomiDEV** [isXander/Controlify#48]
**Is your feature request related to a problem? Please describe.**
"Show Chat" interaction on Steam Deck doesn't make the keyboard show up as well. This leaves the chat box "floating" on top of an empty area (assuming the chat box height was modified, and by the way, the correct configuration value is 0.44625 for Deck, assuming fullscreen gameplay.)

**Describe the solution you'd like**
The idea is to just have the virtual keyboard show up whenever the chat is opened, and hide it whenever the chat is closed. This should ideally account for the scenario in which the keyboard was hidden by the user, to put the chat box back to its original position (just above the hotbar) and clicking on the text box should bring it up again.

Most probably requires to use the Steamworks API to properly function, and IDK if there are Java bindings for that.

**Describe alternatives you've considered**
Clicking STEAM + X after opening chat, but it's a subpar UX.

**Additional context**
Not applicable

**Comment by MIOGMIOG** [comment]
I think this should be also implemented for windows (8+) and linux

**Comment by NyaomiDEV** [comment]
> I think this should be also implemented for windows (8+) and linux (not just steamOS)

This is plain impossible to do. Windows does have a virtual keyboard to show up, but it's only invoked for touch displays and tablet mode, and it would invoke as a separate window IIRC (subpar experience); Linux has a lot of different implementations for virtual keyboards, and it only makes sense to target SteamOS only, since that means targeting Steam Deck and similar consoles specifically.

**Comment by MIOGMIOG** [comment]
> > I think this should be also implemented for windows (8+) and linux (not just steamOS)
> 
> This is plain impossible to do. Windows does have a virtual keyboard to show up, but it's only invoked for touch displays and tablet mode, and it would be invoked as a separate window IIRC (subpar experience); Linux has a lot of different implementations for virtual keyboards (you cannot target them all), and it only makes sense to target SteamOS only, since that means targeting Steam Deck and similar consoles specifically.
> 
> The _only_ way out would be to make a GUI for a keyboard right inside the game, that Controlify uses, but this comes with a lot of drawbacks: for example, Deck users can't use the dual touchpads to input text, which is a downgrade from what SteamOS does offer natively.
> 
> Also, if you put Minecraft as a non-Steam game, theoretically you'd get the Steam virtual keyboard.

Yeah, I mean tablet mode in Windows, it triggers when application wants text input. (But this doesnt work for minecraft)

**Comment by NyaomiDEV** [comment]
> Yeah, I mean tablet mode in Windows, phone-like on screen keyboard opens when application wants text input. (But this doesnt work for minecraft)

It surely requires to have some native libraries built out specifically to support that. Minecraft doesn't have actual text boxes that Windows can see.
